### ISSUE TITLE
Submit error: Not in the workspace

### ISSUE THREAD

**Issue by mikoSL** [exercism/exercism.io#4070]
I tried to submit the hellowork in scala-track

I use Mac and I am in the directory, $pwd  show **/Users/myname/exercism/scala/hello-world**

In this directory, I tried ### exercism submit /Users/myname/exercism/scala/hello-world/src/main/scala/HelloWorld.scala
It showed "Error: not in workspace"

I tried ### exercism submit HelloWorld.scala
It showed "Error, file not found!"

Any suggestion? I am really stuck! 
Thank you for your time.

**Comment by NobbZ** [comment]
Have you tried ``exercism submit src/main/scala/HelloWorld.scala``?

Also, please check the output of ``exercism workspace``. It probably says ``Exercism`` instead of ``exercism``.

The easiest way to fix this is to rename the folder to be with a capital ``E``.

You might need to rename to something temporary first, as I do not now if mac does actually rename if it thinks names are the same.

The reason behind this is that Mac and Windows are case insensitive but case preserving. The CLI though is case sensitive. This causes trouble sometimes and is a known issue.

REF exercism/cli#637

**Comment by mikoSL** [comment]
Thank you for your quick reply. 
It was the problem of capital E. 
Now it is working.

**Comment by kytrinyx** [comment]
We have an open issue to deal with the issues with case sensitivity! Sorry you ran into that one.

**Comment by thefonso** [comment]
Just curious... Why not just make it so lowercase 'e' is the default so Exercism becomes exercism and problem goes away?

**Comment by kytrinyx** [comment]
We're following the pattern set for defaults for each platform. Mac and Windows both default to capitalized directory names, whereas Linux defaults to lowercase.

**Comment by Munodiwa** [comment]
I have the same error, even with the capital letter
C:\Exercism\hello_world.cpp

**Comment by NobbZ** [comment]
@Munodiwa that path is missing track and exercise part. Also please provide the output of `exercism workspace` and the exact message you get when you try to submit something as well as the exact command used an in which directory you are when you enter the command.

**Comment by ask4alikhan** [comment]
> I have the same error, even with the capital letter
> C:\Exercism\hello_world.cpp

I am facing the same issue. getting this error. Even after changing 'E' to 'e'

**Comment by kytrinyx** [comment]
@ask4alikhan The expected path is `$EXERCISM_WORKSPACE/cpp/hello-world/hello_world.cpp` where `$EXERCISM_WORKSPACE` is the output of `exercism workspace`.

If the above isn't enough to sort it out, we're going to need more information in order to help you troubleshoot.

When you submit the file:

1. What directory are you in?
1. What is the exact command that you typed? (Please copy/paste from your terminal)
1. What is the exact error message that is returned? (Please copy/paste from your terminal)
1. What is the output of the command `exercism workspace`?

**Comment by wilhelmlau** [comment]
Sorry to necro this thread but a cursory google search revealed this as the first result (and most relevant). 

I'm currently doing the Javascript track and also on a Mac. 
$pwd 
/Users/me/exercism/javascript/hello-world 
$exercism submit helloworld.js

Error: not in workspace

$exercism workspace will get me an Exercism with a capital E. I realize this is an existing problem, but the thing is that my Exercism folder is already named with a capital E. 


Would appreciate help on other things I could try.

**Comment by NobbZ** [comment]
When you enter the directory, you need to specify it using a capital `E`, so its `cd ~/Exercism/…` rather than `cd ~/exercism/…`.

**Comment by wilhelmlau** [comment]
Oh man quick fix. Thanks.

**Comment by rpottsoh** [comment]
> /Users/me/exercism/javascript/hello-world
> the thing is that my Exercism folder is already named with a capital E.

Based on what you posted it appears to be a little `e`.

Does `exercism workspace` return
```
/Users/me/Exercism
```
or
```
/Users/me/exercism
```
?

**Comment by wilhelmlau** [comment]
@rpottsoh Yeah it was because I entered the directory with a little e. Oof.

**Comment by JMurray90** [comment]
I am typing `/myname/Exercism/hello-world/hello-world.js` and it is still not working. I have tried changing the E capitalization. `exercism workspace` returns `/myname/Exercism`.

**Comment by JMurray90** [comment]
I am getting the following error.
`The file you are trying to submit cannot be found.`

        `/myname/Exercism/hello-world/hello-world.js`

**Comment by NobbZ** [comment]
Hello @JMurray90, can you please go to the folder, where you used to run the tests, and then show the output of `pwd` and `exercism workspace`?

`/myname/Exercism` is not a default workspace path on any of the three supported platforms.

Also, the path you are specifying to the file is missing the language part, which has to be the segment between workspace and exercise name.

Therefore I'd assume a correctly specified full path to your file is one of these:

* `/User/myname/Exercism/javascript/hello-world/hello-world.js` on mac
* `/home/myname/exercism/javascript/hello-world/hello-world.js` on linux

But the easiest way is probably to specify a relative path to the file when you are in the folder where you ran the tests:

```
$ exercism submit hello-world.js
```

**Comment by C-Sharkey** [comment]
> When you enter the directory, you need to specify it using a capital `E`, so its `cd ~/Exercism/…` rather than `cd ~/exercism/…`.

Quick fix worked for me, Thanks!!

**Comment by EvaJR** [comment]
I had this same error on Mac, even though it was working before. Opening a new terminal window and running from the folder where I ran the tests solved it for me.
